docs/doxygen/developer/style.txt

```diff
@@ -0,0 +1,296 @@
+/**
+@page guide_developer_styleguide Style Guide
+@brief Guide for the coding style used in this SDK.
+
+The goal of this style guide is to encourage a readable and consistent coding style across the entire SDK.
+
+@section guide_developer_styleguide_codingstyle Coding Style
+@brief The coding style used in this SDK.
+
+The coding style aims to produce code that is readable and easy to debug. An example is provided in @ref guide_developer_styleguide_codingstyle_example.
+
+@subsection guide_developer_styleguide_codingstyle_generalguidelines General guidelines
+@brief General guidelines for library style.
+- All new libraries (being developed in the development branch) should only use [ANSI C](https://en.wikipedia.org/wiki/ANSI_C#C89) features and standard library functions. (All libraries in the v4_beta branch used the C99 standard, so any changes to the branch can continue using C99).
+- Libraries should log extensively.
+- Code should be well-commented.
+- Only <code>`/`*</code> and <code>*`/`</code> should be used to start and end comments.
+- All comments end with a period.
+- Use of `goto` is discouraged.
+- Only spaces should be used for indenting. A single indent is 4 spaces. No tab characters should be used.
+- A parenthesis is usually followed by a space (see @ref guide_developer_styleguide_codingstyle_example).
+- Lines of code should be less than 80 characters long, although longer lines are permitted.
+- Local variables should be declared at the top of a block in the narrowest scope possible, at the programmer's discretion.
+- All global variables should be declared at the top of a file.
+- Variables are always initialized.
+- A separator is placed between different sections of a file. The current separator is:
+@code{c}
+/*-----------------------------------------------------------*/
+@endcode
+- All files must include the config file at the top of the file <b>before any other includes</b>.
+- `static` functions must have a declaration at the top of the file and be implemented before any application-facing functions.
+- The [GNU complexity](https://www.gnu.org/software/complexity/manual/complexity.html) of every function should be less than 9.
+- Deviations from [MISRA C:2012](https://en.wikipedia.org/wiki/MISRA_C) coding guidelines should be documented as comments in the code.
+
+@subsection guide_developer_styleguide_codingstyle_typeguidelines Type guidelines
+@brief Guidelines for variable types.
+- Only fixed-width integer types should be used. Exceptions for `bool` and types required by third-party APIs.
+- The default integer in the libraries should be 32 bits wide, i.e. `int32_t` or `uint32_t`.
+- Sizes and lengths should be represented with `size_t`, and Boolean variables with `bool`.
```

Comment:
We might want to give more clarification on this. Neither fixed width integers nor bools are a part of C89. Something like a line saying "Although not a part of the C89 standard, libraries may define `bool` macros may be defined for use with C89 compilers."

Reply:
- Although not a part of the C89 standard, only fixed-width integer types should be used. Exceptions for `bool` and types required by third-party APIs.
- The default integer in the libraries should be 32 bits wide, i.e. `int32_t` or `uint32_t`.
- Sizes and lengths should be represented with `size_t`.
- Libraries may define `bool` macros for use with C89 compilers.